core/storage/classifier/gae_models_test.py

```diff
@@ -93,6 +101,51 @@ def test_create_and_get_new_training_job_runs_successfully(self):
         self.assertEqual(training_job.training_data,
                          [{'answer_group_index': 1, 'answers': ['a1', 'a2']}])
 
+    def test_query_training_jobs(self):
+        next_scheduled_check_time = datetime.datetime.utcnow()
+        classifier_models.ClassifierTrainingJobModel.create(
+            'LDAStringClassifier', 'TextInput', 'exp_id1', 1,
+            next_scheduled_check_time,
+            [{'answer_group_index': 1, 'answers': ['a1', 'a2']}],
+            'state_name2', feconf.TRAINING_JOB_STATUS_NEW)
+        classifier_models.ClassifierTrainingJobModel.create(
+            'LDAStringClassifier', 'TextInput', 'exp_id2', 2,
+            next_scheduled_check_time,
+            [{'answer_group_index': 1, 'answers': ['a1', 'a2']}],
+            'state_name2', feconf.TRAINING_JOB_STATUS_PENDING)
+        classifier_models.ClassifierTrainingJobModel.create(
+            'LDAStringClassifier', 'TextInput', 'exp_id3', 3,
+            next_scheduled_check_time + datetime.timedelta(
+                minutes=feconf.CLASSIFIER_JOB_TTL_MINS),
+            [{'answer_group_index': 1, 'answers': ['a1', 'a2']}],
+            'state_name2', feconf.TRAINING_JOB_STATUS_PENDING)
+        classifier_models.ClassifierTrainingJobModel.create(
+            'LDAStringClassifier', 'TextInput', 'exp_id4', 4,
+            next_scheduled_check_time,
+            [{'answer_group_index': 1, 'answers': ['a1', 'a2']}],
+            'state_name2', feconf.TRAINING_JOB_STATUS_FAILED)
+
+        training_jobs, cursor, more = (
+            classifier_models.ClassifierTrainingJobModel.query_training_jobs(
+                None))
+
+        self.assertEqual(len(training_jobs), 5)
```

Comment:
That seems weird. You've created 4 jobs. Why are there 5 jobs coming out?

Edit: Ah, you've got some stuff in setUp(). That's quite confusing, actually. I suggest you break this test out into another test suite. Otherwise it's not clear at all why you're looking at just indices 3 and 4.